### DISCUSSION TITLE
support for ollama

### DISCUSSION THREAD

**Discussion by gtnbssn** [huggingface/llm.nvim#43] [Q&A] [answered]
_Cross-posting this from https://github.com/gsuuon/llm.nvim/discussions/9 which i thought was the same thing!_

---

Does this support [ollama](https://github.com/jmorganca/ollama)?

So far ollama has been my smoothest experience to set up an llm locally.

Here's what the API use looks like (as documented [here](https://ollama.ai/library/phind-codellama)):

```bash
curl -X POST http://localhost:11434/api/generate -d '{
  "model": "phind-codellama",
  "prompt":"Implement a linked list in C++"
 }'
```

**Comment by McPatate** [comment]
I realised that a project with the same name existed after I created it, sorry about that. Hopefully Github's namespacing is enough to differentiate.

I'm discovering the project with your question, so no llm.nvim doesn't support it. I haven't really looked into running models locally yet, though I'd like to get there at some point. I'm going the language server route for this plugin, I initially imagined using [candle](https://github.com/huggingface/candle) to load and run models locally.
Nothing is in the way of supporting multiple backends though, so I'll explore the idea when I get to it, eventually.

**Reply by McPatate** [reply]
Ah but wait I think I understand what you're asking for, it shouldn't be that complicated to support!

I've created an issue: https://github.com/huggingface/llm-ls/issues/17. I'll see depending on my other priorities when I'll tackle it!

I'll start small by just adding the querying to get generations, we'll see later for model creation.

**Reply by eratio08** [reply]
~~Any updates here, as the adapters are now implemented in llm-ls?~~ Thanks for the great work!

Ok nvm just saw https://github.com/huggingface/llm.nvim/pull/72/files?short_path=b335630#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 :D

**Reply by McPatate** [reply, marked as answer]
Ollama is now supported as of `0.5.0`.
